providers/summary/scancode.js

```diff
@@ -90,12 +85,49 @@ class ScanCodeSummarizer {
     }
   }
 
-  _addLicenseFiles(file, declaredLicenses) {
-    // Look for license files at the root of the scanned code
-    // TODO enhance in the future to cover more license management strategies.
-    if (!['license', 'license.txt', 'license.md', 'license.html'].includes(file.path.toLowerCase())) return
-    if (!file.licenses) return
-    file.licenses.forEach(license => declaredLicenses.add(license.spdx_license_key))
+  _summarizeDeclaredLicenseInfo(files) {
+    const declaredLicenses = new Set()
+    for (const fileIdx in files) {
+      const file = files[fileIdx]
+      const pathArray = file.path.split('/')
+      const baseName = last(pathArray)
```

Comment:
If we only check the root we won't be able to get any declared licenses for npm packages. It seems like all the npm packages have their license file inside the package directory. Should we make this specific exception for npm? Along with that, in the future would we have to make these exceptions for each type of package manager depending on how they structure their projects?

Reply:
If it is an NPM that we are scanning then the License file (if any) is of secondary importance. Since NPM (and typical other package managers) have explicit metadata to call out the license (e.g, entry in the package.json file), we should be using that information as the declared license. So this method returning nothing is fine as a first step. `summarizePackageInfo` should pick up the license in the metadata.

We may need to revise this approach but lets try the simple thing and see how it goes.